### DISCUSSION TITLE
Need help with shortcuts

### DISCUSSION THREAD

**Discussion by SpidFightFR** [nix-community/plasma-manager#424] [Q&A] [answered]
Hey there, sorry i spam the discussions (i hope these will help newcomers afterwards):

I'm trying to set-up a shortcut for Krunner (as an example).

But i struggle to understand the inputs to insert in the options, so far github copilot (:skull:) says it needs to be something along these lines:

```nix
programs.plasma.shortcuts = {
  "applications/krunner" = {
    "_launch" = ["Meta+D"];
    };
  };
```

But it doesn't seem to work.
Should i put `org.kde.krunner.desktop` as the app name instead ?
and/or should i change the "launch" option ? (to execute krunner, default is ALT + SPACE)

**Comment by HeitorAugustoLN** [comment, marked as answer]
Generally you can just see what is being changed in their respective configurations. For example, shortcuts are defined in `kglobalshortcutsrc`. So every change plasma in there, is just what will be written by plasma-manager.

Shortcuts for programs such as KRunner, Dolphin, Neovim, etc. Are defined in the following pattern:

```
[services][desktop_file.desktop]
```

Krunner, for example, is:

```
[services][org.kde.krunner.desktop]
```

And when there are nested configuration groups like this. You need to set them in plasma-manager with a forward slash, like this: `services/org.kde.krunner.desktop`. So the way you would set it using the shortcuts module is by writing:

```nix
{
  programs.plasma.shortcuts = {
    "services/org.kde.krunner.desktop" = {
      _launch = "Meta+D";
    };
  };
}
```

If you wanted to use multiple shortcuts for krunner, it would be almost the same thing:

```nix
{
  programs.plasma.shortcuts = {
    "services/org.kde.krunner.desktop" = {
      _launch = ["Meta+D" "Alt+Space"];
    };
  };
}
```

**Reply by SpidFightFR** [reply]
I see, thanks again !
